gems/sorbet-runtime/lib/types/enum.rb

```diff
@@ -90,6 +90,7 @@ def self.try_deserialize(serialized_val)
   # @raise [KeyError] if serialized value does not match any instance.
   sig {overridable.params(serialized_val: SerializedVal).returns(T.attached_class).checked(:never)}
   def self.from_serialized(serialized_val)
+    return serialized_val if serialized_val.is_a?(self)
```

Comment:
This seems wrong. Why is it ok for `from_serialized` to get something that is not the output of `serialize`?

At the very least, we should change the sig (here, and in the RBI) to make it clear that `from_serialized` is sometimes called with `T.attached_class`.

Reply:
On reflection, you're right, this is the wrong place. The problem here is in `#with`: I've changed the approach to modify that.